### PR TITLE
hotspot veto

### PR DESCRIPTION
**Pull request by LuisSanchez25**
I think this might be the first a few time dependent cuts people will want to add to xedocs, should we make a separate folder for these cuts since they are not really corrections?

### CODE REVIEW

**Review by jmosbacher — COMMENTED** [review]
Wondering if "corrections" is the correct category for this schema. The corrections are immutable, are we sure this is needed?
Also, take a look at the hotspot report schema under "operations_reports", are you sure this cant be used for cuts?

**Comment by LuisSanchez25** [comment]
From my understanding with what Kexin was trying to do it seems like she wanted to have an individual value to be used for a certain number of runs (those with hotspots). I can ask for more details/clarification to see if she does want these values to be immutable and make new version in the future or not.
Looking at the hotspot file under "operations_reports", I dont think it covers what she had in mind but Ill talk to her.

**Comment by jmosbacher** [comment]
@LuisSanchez25 I suggest that you invite the analysts that requested this schema to participate in the discussion here so we have a proper record of the development.

**Comment by LuisSanchez25** [comment]
@chnlkx

**Comment by chnlkx** [comment]
Hi Luis and Yossi! Thanks for starting this PR : )
- As Luis mentioned, what we trying to put into the time-dependent corrections is a series of single values of the hotspot_veto_cut threshold for each run throughout the science run. These values would be immutable and versioned.
- And for the hotspot file under "operations_reports", from the content I think it's for the operation log purpose, maintained by the shifters, and only involves a small number of runs. Yet the one we are trying to set up is an outcome of the offline analysis and would involve all the science data runs. So from my understanding, they are not suitable to combine due to different usage scenarios.

**Comment by jmosbacher** [comment]
@chnlkx thanks for the extra details. So if i understand you correctly, this collection will store a threshold number that will be used to cut data that is above this threshold? what config in the cut plugin will this value be used for?
If this is the case, I suggest renaming the schema to something more suitable to its use and adding some more info in the module/class docstring

File: xedocs/schemas/corrections/hotspot_veto_cut.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+The 'hotspot_veto_threshold' provides a per-run cut threshold of the 'hotspot_veto_cut',
+which vetos the events with local single electron rate above the threshold in each run.
+Reference: xenon:xenonnt:analysis:hot_spot_cut_summary
+"""
+
+from .base_corrections import TimeIntervalCorrection
+
+class HotspotVetoThreshold(TimeIntervalCorrection):
+    _ALIAS = 'hotspot_veto_threshold'
```
Comment (jmosbacher) [inline xedocs/schemas/corrections/hotspot_veto_cut.py:10]:
```suggestion
    _ALIAS = 'hotspot_veto_thresholds'
```
By convention we use the plural for the name of the collection of documents.

**Review by jmosbacher — APPROVED** [review]
I think this looks fine.

**Review by jmosbacher — APPROVED** [review]

File: xedocs/schemas/corrections/hotspot_veto_cut.py (hunk 1 above)
Comment (chnlkx) [reply xedocs/schemas/corrections/hotspot_veto_cut.py]: OK thanks! I'll also adjust the cutax plugin accordingly.😊

**Pull request merged**